### ISSUE TITLE
Allow address.parse() to reject addresses with local hostnames

### ISSUE THREAD

**Issue by grvsmth** [mailgun/flanker#201]
`address.parse()` currently accepts addresses with a single-part domain as used for a local mail host, e.g. `bleah@example`.  I am currently working on a use case where we would only want to send to Internet addresses, like `bleah@example.com` or `bleah@foo.example.com`.  It would be useful to have an option to reject addresses with single-part domains.

**Comment by horkhe** [comment]
It seems to be too use case specific to be included as a parser option. I can think of many similar specific use cases e.g.: do not allow domains of 3rd, 4th, Xth level; do not allow domains from specific top level domains; do not allow domains from a certain geo region... All that seems like a higher level of abstraction that lays out of the scope of email address syntax parsing.

In your case the solution is as simple as:

**Comment by b0d0nne11** [comment]
I would agree. There maybe other users of the library that have a valid use case to send to `bob@localhost` for example.

**Comment by grvsmth** [comment]
To clarify, Brendan, I am not suggesting we remove the validation of addresses like `blah@example`.  I am suggesting we add the option to filter out addresses with that form.

**Comment by grvsmth** [comment]
Thanks for your feedback, Maxim.  I do not see the other use cases you mention as being anywhere near as common, and I hope you will consider adding this option.  In the meantime I will add a check in my script along the lines you suggest.

**Comment by grvsmth** [comment]
We don't really want to go to the trouble of calling `validate_address()` when all we want is to check the form of the address, but thanks.
